lib/widgets/app.dart

```diff
@@ -129,7 +135,11 @@ class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
                 (_) => widget._declareReady());
             }
             GlobalLocalizations.zulipLocalizations = ZulipLocalizations.of(context);
-            return child!;
+            final locale = Localizations.localeOf(context); // Get locale from context
```

Comment:
nit: this comment says only information that's equally readable from the code itself, so just leave it out:
```suggestion
            final locale = Localizations.localeOf(context);
```